src/tcgwars/logic/impl/gen4/SupremeVictors.groovy

```diff
@@ -229,378 +245,518 @@ public enum SupremeVictors implements LogicCardInfo {
+                opp.all.findAll{it.types.containsAny(defending.types)}.each {
+                  damage 20, it
+                }
+              }
             }
           }
-
         };
       case RAYQUAZA_C_8:
-        return basic (this, hp:HP100, type:COLORLESS, retreatCost:3) {
+        return basic (this, hp:HP100, type:C, retreatCost:3) {
           weakness C
           resistance F, MINUS20
           move "Trash Burst", {
             text "10+ damage. You may discard up to 5 Energy cards from your hand. If you do, this attack does 10 damage plus 10 more damage for each Energy card you discarded."
             energyCost C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 10
+              def energyList = my.hand.select(min:0,max:5,"You may discard up to 5 Energy cards in order to deal 10 additional damage per Energy card",cardTypeFilter(ENERGY))
+              damage 10 * energyList.size()
+              afterDamage {
+                energyList.discard()
+              }
             }
           }
           move "Twister", {
             text "50 damage. Flip 2 coins. If both are tails, this attack does nothing. For each heads, discard an Energy attached to the Defending Pokémon."
             energyCost C, C, C, C
             attackRequirement {}
             onAttack {
-              damage 0
+              flip 2, {}, {}, [
+                2: {
+                  damage 50
+                  discardDefendingEnergyAfterDamage C, C
+                },
+                1: {
+                  damage 50
+                  discardDefendingEnergyAfterDamage C
+                }]
             }
           }
-
         };
       case REGIGIGAS_FB_9:
-        return basic (this, hp:HP100, type:COLORLESS, retreatCost:4) {
+        return basic (this, hp:HP100, type:C, retreatCost:4) {
           weakness F
           move "Drain Punch", {
-            text "30 damage. a number of damage counters equal to the amount of Energy attached to the Defending Pokémon."
+            text "30 damage. Remove from Regigigas FB a number of damage counters equal to the amount of Energy attached to the Defending Pokémon."
             energyCost C, C, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 30
+              afterDamage {
+                heal 10 * defending.card.energyCount(C), self
+              }
             }
           }
           move "Rainbow Lariat", {
-            text "20× damage. you have in play."
+            text "20x damage. Does 20 damage times the number of different types of Pokémon SP you have in play."
             energyCost C, C, C, C
             attackRequirement {}
             onAttack {
-              damage 0
+              for(Type t1:Type.values()){
+                if(my.bench.find{it.pokemonSP && it.types.contains(t1)})
```

Comment:
This attack mentions different types of Pokemon SP, maybe create an array each time damage is done, and for each subsequent iteration, check if it's in the array before doing damage

Reply:
I don't understand this one. It's already iterating over all the types and if a pokemonSP with that type is on the bench it adds damage which seems right to me. It should also check the active spot too since Regigigas is also a pokemonSP though.

Reply:
Ah that makes sense.  The way it is currently coded is correct then, except for `my.bench` which I agree should be `my.all`